### PR TITLE
Feat Lock App - Pin/Fingerprint

### PR DESCRIPTION
**Pull request by JDonadio**
TODO:

- [x] Styling for Copay
- [x] Styling for BitPay
- [x] Locked view
- [x] Messages
- [x] Fingerprint
- [x] Request -> Resume event
- [x] Attempts limit control -> timer
- [x] Avoid to lock without backup

Related feature https://github.com/bitpay/copay/issues/4555

### CODE REVIEW

File: www/views/deadview.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<ion-view id="dead-view">
+  <ion-nav-bar class="bar-royal">
+    <ion-nav-title>{{title}}</ion-nav-title>
```
Comment (gabrielbazan7) [inline www/views/deadview.html]: I think we don't need a nav bar and a title here

File: www/views/pincode.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
+<ion-view id="pin-code" ng-controller="pincodeController" hide-tabs hide-back-button="!fromSettings">
+  <ion-nav-bar class="bar-clear">
+    <ion-nav-back-button>
+    </ion-nav-back-button>
+  </ion-nav-bar>
+  <div class="content">
+    <div class="block-text row">
+      <span translate ng-if="!newPincode && !isComplete()">Please enter your mobile unlock code</span>
+      <span translate ng-if="newPincode && !isComplete()">Confirm your mobile unlock code</span>
```
Comment (gabrielbazan7) [inline www/views/pincode.html]: "Verify your code" or "Verify your passcode"

File: www/views/pincode.html [hunk 3]
```diff
@@ -0,0 +1,72 @@
+<ion-view id="pin-code" ng-controller="pincodeController" hide-tabs hide-back-button="!fromSettings">
+  <ion-nav-bar class="bar-clear">
+    <ion-nav-back-button>
+    </ion-nav-back-button>
+  </ion-nav-bar>
+  <div class="content">
+    <div class="block-text row">
+      <span translate ng-if="!newPincode && !isComplete()">Please enter your mobile unlock code</span>
```
Comment (gabrielbazan7) [inline www/views/pincode.html]: "Enter your code" or "Enter your passcode"

**Comment by matiu** [comment]
Great work!

Some comments:

[ ] - Entering the 4th digit should "enter" the pin afterwards (no need to the "check" button)
[ ] - Remove the "check" button on /pin
[ ] - Incorrect PIN does not reset the input

**Comment by JDonadio** [comment]
Updated. Thanks for your comments!

File: www/views/pin.html [hunk 4]
```diff
@@ -0,0 +1,73 @@
+<ion-view id="pin" hide-tabs hide-back-button="!fromSettings">
+  <ion-nav-bar class="bar-clear">
+    <ion-nav-back-button>
+    </ion-nav-back-button>
+  </ion-nav-bar>
+  <div class="content">
+    <div class="block-text row">
+      <div class="message" ng-if="!confirmPin && !error" translate>Please enter your unlock PIN</div>
```
Comment (gabrielbazan7) [inline www/views/pin.html]: I think the text should say "Please enter your PIN" and "Confirm your PIN"

File: www/views/lockedView.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<ion-view id="locked-view">
+  <ion-nav-bar class="bar-royal">
+    <ion-nav-title>{{title}}</ion-nav-title>
```
Comment (gabrielbazan7) [inline www/views/lockedView.html]: I think you should remove the nav bar. This title is not necessary. And also is wrong. Should be a title for the view, not a state.

**Comment by matiu** [comment]
On Lock controller.

UI should be a: a radio controller:
 * Disabled
 * Lock app using PIN
 * Lock app using fingerprint.

**Comment by JDonadio** [comment]
Updated. Thanks for the detailed review and for all the comments!

**Pull request merged**